core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceLocal.java

```diff
@@ -51,4 +52,14 @@ public int getAccountIndex() {
 	public void setAccountIndex(int index) {
 		blockchainSharedPreferences.edit().putInt(ACCOUNT_INDEX_KEY, index).apply();
 	}
+
+	@Override
+	public void setAppId(@NonNull String appId) {
+		blockchainSharedPreferences.edit().putString(APP_ID_KEY, appId).apply();
+	}
+
+	@Override
+	public String getAppId() {
```

Comment:
why do we need to store app_id in blockchain? - it is better to hold a single point of truth